mir_eval/alignment.py

```diff
@@ -0,0 +1,300 @@
+    "LyricSynchronizer: Automatic synchronization system between musical audio signals and lyrics",
+    IEEE Journal of Selected Topics in Signal Processing, VOL. 5, NO. 6, 2011
+
+"""
+
+import collections
+
+import numpy as np
+from scipy.stats import skewnorm
+
+from mir_eval.util import filter_kwargs
+
+
+def validate(reference_timestamps: np.ndarray, estimated_timestamps: np.ndarray):
+    """Checks that the input annotations to a metric look like valid onset time
+    arrays, and throws helpful errors if not.
+    Parameters
+    ----------
+    reference_timestamps : np.ndarray
+        reference timestamp locations, in seconds
+    estimated_timestamps : np.ndarray
+        estimated timestamp locations, in seconds
+    """
+    # We need to have 1D numpy arrays
+    if not isinstance(reference_timestamps, np.ndarray):
+        raise ValueError(
+            f"Reference timestamps need to be a numpy array, but got {type(reference_timestamps)}"
+        )
+    if not isinstance(estimated_timestamps, np.ndarray):
+        raise ValueError(
+            f"Estimate timestamps need to be a numpy array, but got {type(estimated_timestamps)}"
+        )
+    if reference_timestamps.ndim != 1:
+        raise ValueError(
+            "Reference timestamps need to be a one-dimensional vector, but got"
+            f" {reference_timestamps.ndim} dimensions"
+        )
+    if estimated_timestamps.ndim != 1:
+        raise ValueError(
+            "Estimate timestamps need to be a one-dimensional vector, but got"
+            f" {estimated_timestamps.ndim} dimensions"
+        )
+
+    # If reference or estimated timestamps are empty, cannot compute metric
+    if reference_timestamps.size == 0:
+        raise ValueError("Reference timestamps are empty.")
+    if estimated_timestamps.size != reference_timestamps.size:
+        raise ValueError(
+            "Number of timestamps must be the same in prediction and ground truth, "
+            f"but found {estimated_timestamps.size} in prediction and "
+            f"{reference_timestamps.size} in ground truth"
+        )
+
+    # Check monotonicity
+    if not np.all(reference_timestamps[1:] - reference_timestamps[:-1] >= 0):
+        raise ValueError("Reference timestamps are not monotonically increasing!")
+    if not np.all(estimated_timestamps[1:] - estimated_timestamps[:-1] >= 0):
+        raise ValueError("Estimate timestamps are not monotonically increasing!")
+
+    # Check positivity (need for correct PCS metric calculation)
+    if not np.all(reference_timestamps >= 0):
+        raise ValueError("Reference timestamps can not be below 0!")
+    if not np.all(estimated_timestamps >= 0):
+        raise ValueError("Estimate timestamps can not be below 0!")
+
+
+def ae(reference_timestamps, estimated_timestamps):
+    """Compute the absolute deviations between predicted and reference timestamps,
+    and then takes the median or average over all events
```

Comment:
"takes the median or average over all events" -> "return the median and average over all events."